plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java

```diff
@@ -294,6 +300,19 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
                     }
                     if (markerSeverity != -1) {
                         line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2))) - 1;
+                        String lineContents = PySelection.getLine(document, line);
```

Comment:
I know this works, but I think we can be a bit less strict about it in this case.

You can just use: `org.python.pydev.core.CheckAnalysisErrors.isCodeAnalysisErrorHandled(lineContents, "type")` for this.